main.py

```diff
@@ -164,10 +165,27 @@ def is_neutral(status):
     query_result = data["query_result"]
     data = query_result["data"]
     rows = data["rows"]
+
+    # Determine the date range of this dataset.
+    min_timestamp = float("inf")
+    max_timestamp = -1
```

Comment:
Why not just...?
```python
min_timestamp = min(x['min_timestamp'] for x in rows)
max_timestamp = max(x['max_timestamp'] for x in rows)
```

Reply:
hehe, true. I first put it within the main loop and moved it above like an animal to get the printout first :D